stdlib/public/core/FloatingPointParsing.swift.gyb

```diff
@@ -126,36 +126,23 @@ extension ${Self}: LosslessStringConvertible {
   ///   is `nil`.
   @inlinable // FIXME(sil-serialize-all)
```

Comment:
thinking of this comment (that we should probably remove...) could we use `@_specialize` and then make this non-inlinable to reduce having to inline it?

Reply:
given there are only 2 string protocol conformance rn

Reply:
Not unreasonable.

Reply:
Even better, make the generic thing a stub that just canonicalizes to `Substring` and make the non-inlinable implement take `Substring`. (Which is almost always better in every respect than using `StringProtocol`)

Reply:
Do you mean like this?
```
  @inlinable
  public init?(_ text: String) {
    self.init(substring: text[...])
  }

  public init?(substring: Substring) {
    ... real work ...
  }
```

Reply:
Yeah.

Reply:
Done.